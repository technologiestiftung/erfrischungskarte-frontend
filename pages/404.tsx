import ErrorPage from '@components/ErrorPage'
import { FC } from 'react'

export const FourOFour: FC = () => (
  <ErrorPage statusCode={400} message="Diese Seite wurde nicht gefunden" />
)

export default FourOFour
