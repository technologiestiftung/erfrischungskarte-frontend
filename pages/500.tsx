import ErrorPage from '@components/ErrorPage'
import { FC } from 'react'

export const FiveHundred: FC = () => (
  <ErrorPage statusCode={500} message="Interne Server Fehler" />
)

export default FiveHundred
