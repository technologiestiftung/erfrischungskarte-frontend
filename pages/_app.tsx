import { mapRawQueryToState } from '@lib/utils/queryUtil'
import { RefreshmentMap } from '@modules/RefreshmentMap'
import { ParsedUrlQuery } from 'querystring'
import { StrictMode, FC } from 'react'
import { Head } from '@components/Head'
import '../src/style/global.css'
import '../src/components/MapControls/mapControls.css'
import '../src/components/MapPoiTooltip/MapPoiTooltip.css'
import { useMatomo } from '@lib/hooks/useMatomo'

interface PagePropType extends Record<string, unknown> {
  title?: string
  query: ParsedUrlQuery
}

interface ComponentPropType {
  title?: string
  query?: ReturnType<typeof mapRawQueryToState>
}

const App: FC<{
  Component: FC<ComponentPropType>
  pageProps: PagePropType
}> = ({ Component, pageProps }) => {
  useMatomo()
  const parsedQuery = pageProps.query ? mapRawQueryToState(pageProps.query) : {}

  return (
    <StrictMode>
      <RefreshmentMap {...pageProps} query={parsedQuery}>
        <Head pageTitle={pageProps.title || ''} />
        <Component {...pageProps} query={parsedQuery} />
      </RefreshmentMap>
    </StrictMode>
  )
}

export default App
