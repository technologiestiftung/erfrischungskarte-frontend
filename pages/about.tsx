import { Accordion } from '@components/Accordion'
import { CitylabLogo } from '@components/CitylabLogo'
import { OdisLogo } from '@components/OdisLogo'
import { SenWebLogo } from '@components/SenWebLogo'
import { TsbLogo } from '@components/TsbLogo'
import {
  ABOUT_ACCORDION_ITEMS,
  ABOUT_CONTACT_TEXT,
  ABOUT_INTRODUCTION_TEXT,
} from '@modules/RefreshmentMap/content'
import classNames from 'classnames'
import { GetServerSideProps } from 'next'
import React, { FC } from 'react'

// eslint-disable-next-line @typescript-eslint/require-await
export const getServerSideProps: GetServerSideProps = async ({ query }) => ({
  props: {
    title: 'Über das Projekt',
    query,
  },
})

const focusStyles =
  'focus:outline-none focus:ring-2 focus:ring-gray-800 focus:ring-offset-2 focus:rounded-sm focus:ring-offset-gray-100'
export const About: FC = () => (
  <div>
    <p className="text-gray-500 text-sm pb-6">{ABOUT_INTRODUCTION_TEXT}</p>
    <Accordion items={ABOUT_ACCORDION_ITEMS} />
    <p className="text-gray-500 text-sm pt-6">{ABOUT_CONTACT_TEXT}</p>
    <section className="mt-16 flex flex-wrap">
      <div className="flex flex-col mr-6 mb-6">
        <span className="text-sm mb-2">Ein Projekt der</span>
        <TsbLogo className={focusStyles} />
      </div>
      <div className="flex flex-col mb-6">
        <span className="text-sm mb-2">Durchgeführt von der</span>
        <a
          href="https://odis-berlin.de/"
          target="_blank"
          rel="noopener noreferrer"
          aria-label="Open Data Informationsstelle Berlin"
          className={focusStyles}
        >
          <OdisLogo className={`w-40`} />
        </a>
      </div>
      <div className="flex flex-col mb-6">
        <span className="text-sm mb-2">In Zusammenarbeit mit dem</span>
        <a
          href="https://www.citylab-berlin.org"
          target="_blank"
          rel="noopener noreferrer"
          aria-label="CityLAB Berlin"
          className={focusStyles}
        >
          <CitylabLogo className={`w-36`} />
        </a>
      </div>
      <div className="flex flex-col">
        <span className="text-sm mb-2">Gefördert von</span>
        <SenWebLogo className={focusStyles} />
      </div>
    </section>
    <footer
      className={classNames(
        'mt-8 bg-gray-100 p-8 -ml-6 -mb-6 sm:-ml-8 sm:-mb-8',
        'flex flex-wrap'
      )}
      style={{
        width: 'var(--sidebarWidth, 320px)',
      }}
    >
      <span className="text-xs w-full mb-4">
        © 2021 Technologiestiftung Berlin
      </span>
      <a
        href="https://www.technologiestiftung-berlin.de/de/impressum/"
        className={`text-xs hover:underline ${focusStyles} mr-4`}
        target="_blank"
        rel="noreferrer"
      >
        Impressum
      </a>
      <a
        href="https://www.technologiestiftung-berlin.de/de/datenschutz/"
        className={`text-xs hover:underline ${focusStyles}`}
        target="_blank"
        rel="noreferrer"
      >
        Datenschutzerklärung
      </a>
    </footer>
  </div>
)

export default About
