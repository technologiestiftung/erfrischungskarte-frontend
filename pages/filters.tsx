import { mapRawQueryToState } from '@lib/utils/queryUtil'
import { GetServerSideProps } from 'next'
import React, { FC } from 'react'
import { FilterChip } from '@components/FilterChip'
import { PoiLegendItem } from '@components/PoiLegendItem'
import {
  LAYER_LEGEND_ITEMS,
  PoiCategory,
  POI_CATEGORY_COLOR_MAP,
  POI_CATEGORY_ID_MAP,
  SHADE_SUPPORT_NOTE,
} from '@modules/RefreshmentMap/content'
import { useHasMobileSize } from '@lib/hooks/useHasMobileSize'
import { useHasWebPSupport } from '@lib/hooks/useHasWebPSupport'
import classNames from 'classnames'
import { LayerLegendBlock } from '@components/LayerLegendBlock'
import { Warning } from '@components/Warning'
import { useRouter } from 'next/router'

// eslint-disable-next-line @typescript-eslint/require-await
export const getServerSideProps: GetServerSideProps = async ({ query }) => ({
  props: {
    title: 'Filter',
    query,
  },
})

const poiCategoryColorArray: [PoiCategory, string][] = Array.from(
  POI_CATEGORY_COLOR_MAP
)

const defaultActivePoiIds = Object.values(POI_CATEGORY_ID_MAP)

export const Filters: FC<{
  query: ReturnType<typeof mapRawQueryToState>
}> = () => {
  const hasMobileSize = useHasMobileSize()
  const hasWebPSupport = useHasWebPSupport()
  const { query, replace: routerReplace } = useRouter()
  const mappedQuery = mapRawQueryToState(query)

  const {
    shade: shadeLegendContent,
    temperature: temperatureLegendContent,
    wind: windLegendContent,
  } = LAYER_LEGEND_ITEMS

  const windAndTemperatureFiltersAreActivated =
    (!Object.keys(mappedQuery).includes('showWind') &&
      !Object.keys(mappedQuery).includes('showTemperature')) ||
    (mappedQuery.showWind === true && mappedQuery.showTemperature === true)

  return (
    <div className="grid grid-cols-1">
      <section
        aria-label="Angezeigte Orte auswählen"
        className={classNames(
          'col-span-1',
          hasMobileSize ? 'row-start-2 mt-4' : 'row-start-1'
        )}
      >
        {!hasMobileSize && (
          <>
            <h2 className="text-lg font-bold">Orte</h2>
            <p className="text-gray-500">
              Wähle aus, welche Orte zum Verweilen du auf der Karte sehen
              möchtest.
            </p>
          </>
        )}
        <div
          className={classNames(
            'flex flex-wrap',
            hasMobileSize ? 'mt-0' : 'mt-4'
          )}
        >
          {poiCategoryColorArray.map(([category, color]) => {
            const poiId = POI_CATEGORY_ID_MAP[category]
            return (
              <div key={category} className="mt-2 mr-2">
                <FilterChip
                  ariaLabel={category}
                  isSelected={mappedQuery.places?.includes(poiId)}
                  handleClick={() => {
                    let activePlaces

                    const noPlacesInQuery = !mappedQuery.places

                    const placesValueIsZero =
                      mappedQuery.places && mappedQuery.places[0] === 0

                    const onlyOnePlaceInQuery =
                      mappedQuery.places && mappedQuery.places.length === 1

                    const clickedPlaceAlreadyInQuery =
                      mappedQuery.places && mappedQuery.places?.includes(poiId)

                    const clickedPlaceNotYetInQuery =
                      mappedQuery.places && !mappedQuery.places.includes(poiId)

                    if (noPlacesInQuery) {
                      activePlaces = defaultActivePoiIds.filter(
                        (defaultPoiId) => defaultPoiId !== poiId
                      )
                    }

                    if (clickedPlaceNotYetInQuery) {
                      if (placesValueIsZero) {
                        activePlaces = [poiId] // activates clicked place after all places were deactivated before
                      } else {
                        activePlaces = mappedQuery.places?.concat(poiId) // activates clicked place
                      }
                    }

                    if (clickedPlaceAlreadyInQuery) {
                      if (onlyOnePlaceInQuery && !placesValueIsZero) {
                        activePlaces = [0] // deactivates all places
                      } else {
                        activePlaces = mappedQuery.places?.filter(
                          (placeId) => placeId !== poiId // deactivates clicked place
                        )
                      }
                    }

                    void routerReplace(
                      {
                        query: {
                          ...mappedQuery,
                          places: activePlaces,
                        },
                      },
                      undefined
                    )
                  }}
                >
                  <PoiLegendItem label={category} color={color} />
                </FilterChip>
              </div>
            )
          })}
        </div>
      </section>
      <section
        aria-label="Angezeigte Flächen auswählen"
        className={classNames(
          'col-span-1',
          hasMobileSize ? 'row-start-1' : 'row-start-2',
          hasMobileSize ? 'pb-6' : 'mt-14',
          hasMobileSize ? 'border-b border-gray-400 border-dotted' : 'border-0'
        )}
      >
        {!hasMobileSize && (
          <>
            <h2 className="text-lg font-bold">Flächen</h2>
            <p className="text-gray-500">
              Wähle aus, welche Art von erfrischenden Bereichen du sehen
              möchtest.
            </p>
          </>
        )}
        {shadeLegendContent && (
          <div className={classNames(hasMobileSize ? 'mt-1' : 'mt-6')}>
            <LayerLegendBlock
              title={shadeLegendContent.title}
              description={shadeLegendContent.description}
              icon={shadeLegendContent.icon}
              legendFigure={
                hasWebPSupport ? (
                  shadeLegendContent.legendFigure
                ) : (
                  <Warning>{SHADE_SUPPORT_NOTE}</Warning>
                )
              }
              layerIsActive={mappedQuery.showShadows !== false}
              layerIsDisabled={!hasWebPSupport}
              handleToggle={() => {
                void routerReplace(
                  {
                    query: {
                      ...mappedQuery,
                      showShadows: Object.keys(mappedQuery).includes(
                        'showShadows'
                      )
                        ? !mappedQuery.showShadows
                        : false,
                    },
                  },
                  undefined
                )
              }}
            />
          </div>
        )}
        {temperatureLegendContent && (
          <div className={classNames(hasMobileSize ? 'mt-4' : 'mt-6')}>
            <LayerLegendBlock
              title={temperatureLegendContent.title}
              description={temperatureLegendContent.description}
              icon={temperatureLegendContent.icon}
              legendFigure={temperatureLegendContent.legendFigure}
              layerIsActive={mappedQuery.showTemperature !== false}
              handleToggle={() => {
                void routerReplace(
                  {
                    query: {
                      ...mappedQuery,
                      showTemperature: Object.keys(mappedQuery).includes(
                        'showTemperature'
                      )
                        ? !mappedQuery.showTemperature
                        : false,
                    },
                  },
                  undefined
                )
              }}
            />
          </div>
        )}
        {windLegendContent && (
          <div className={classNames(hasMobileSize ? 'mt-4' : 'mt-6')}>
            <LayerLegendBlock
              title={windLegendContent.title}
              description={windLegendContent.description}
              icon={windLegendContent.icon}
              legendFigure={windLegendContent.legendFigure}
              layerIsActive={mappedQuery.showWind !== false}
              handleToggle={() => {
                void routerReplace(
                  {
                    query: {
                      ...mappedQuery,
                      showWind: Object.keys(mappedQuery).includes('showWind')
                        ? !mappedQuery.showWind
                        : false,
                    },
                  },
                  undefined
                )
              }}
            />
          </div>
        )}
        {windAndTemperatureFiltersAreActivated && (
          <p
            className={classNames(
              'text-xs',
              'mt-4',
              hasMobileSize ? 'ml-10' : null
            )}
          >
            <strong className="italic text-layer-turquoise-300">
              Aufgepasst:
            </strong>
            {` `}
            {`Gleichzeitig kühle und windige Flächen werden ${
              !hasMobileSize ? 'durch die Überlagerung von Blau und Grün' : ''
            } Türkis dargestellt.`}
          </p>
        )}
      </section>
    </div>
  )
}

export default Filters
