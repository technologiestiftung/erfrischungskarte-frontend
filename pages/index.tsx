import { FC } from 'react'
import { RefreshmentMap } from './../src/modules/RefreshmentMap'

export const Home: FC = () => <RefreshmentMap />

export default Home
