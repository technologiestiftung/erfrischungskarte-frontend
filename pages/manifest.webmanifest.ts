import { NextApiResponse } from 'next'
import { Component } from 'react'
import colors from '../src/style/colors'

const getManifest = ({
  siteTitle = 'Berliner Erfrischungskarte',
  themeTextColor = colors['layer-turquoise']['300'],
  themeBackgroundColor = '#ffffff',
}): string => `{
    "name": "${siteTitle}",
    "short_name": "FreshMap",
    "icons": [
        {
            "src": "/favicons/android-chrome-192x192.png",
            "sizes": "192x192",
            "type": "image/png"
        },
        {
            "src": "/favicons/android-chrome-512x512.png",
            "sizes": "512x512",
            "type": "image/png"
        }
    ],
    "theme_color": "${themeTextColor}",
    "background_color": "${themeBackgroundColor}",
    "display": "standalone"
}
`

class Sitemap extends Component {
  // eslint-disable-next-line @typescript-eslint/require-await
  static async getInitialProps({
    res,
  }: {
    res: NextApiResponse
  }): Promise<void> {
    res.setHeader('Content-Type', 'application/manifest+json')
    res.write(getManifest({}))
    res.end()
  }
}

export default Sitemap
