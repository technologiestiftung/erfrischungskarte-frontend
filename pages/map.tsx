import { GetServerSideProps } from 'next'
import { FC } from 'react'

// eslint-disable-next-line @typescript-eslint/require-await
export const getServerSideProps: GetServerSideProps = async ({ query }) => ({
  props: {
    title: 'Karte',
    query,
  },
})

export const Map: FC = () => null

export default Map
