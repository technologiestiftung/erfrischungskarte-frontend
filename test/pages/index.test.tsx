import React from 'react'
import { render } from '../testUtils'
import { Home } from '../../pages/index'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
