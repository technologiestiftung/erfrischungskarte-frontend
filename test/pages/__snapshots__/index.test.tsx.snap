// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="position: relative; width: 800px; height: 500px; cursor: grab; outline: none;"
    tabindex="0"
  >
    <div
      style="position: relative; width: 100%; height: 100%;"
    >
      <div
        class=""
        style="position: absolute; width: 100%; height: 100%; overflow: hidden; visibility: inherit;"
      />
    </div>
  </div>
</DocumentFragment>
`;
